app/src/main/java/com/infomaniak/drive/ui/publicShare/PublicShareViewModel.kt

```diff
@@ -113,6 +125,41 @@ class PublicShareViewModel(val savedStateHandle: SavedStateHandle) : ViewModel()
         // TODO: Manage apiResponse when the backend will be done
     }
 
+    fun executeDownloadAction(
+        activityContext: Context,
+        downloadAction: DownloadAction,
+        file: File?,
+        navigateToDownloadDialog: suspend () -> Unit,
+        onDownloadSuccess: () -> Unit,
+        onDownloadError: () -> Unit,
+    ) = viewModelScope.launch(Dispatchers.IO) {
```

Comment:
The limit of parameters for function is not 5 ? We might create a data class here maybe ?

Reply:
The sonar cloud limit is 7